client/src/partials/patient_invoice/patientInvoice.js

```diff
@@ -155,6 +159,22 @@ function PatientInvoiceController(Patients, PatientInvoices, PatientInvoiceForm,
 
     if (angular.isDefined(vm.services) && vm.services.length) {
       vm.Invoice.setService(vm.services[SERVICE_INDEX]);
+      vm.selectedService = vm.services[SERVICE_INDEX];
+    }
+  }
+
+  function setService (service_id) {
+
+    Services.read(service_id)
```

Comment:
This makes a `GET` request every time a service is selected, usually this will result in at least 2 additional requests per page use. 

All the information has already been downloaded by this controller with `Services.read()`, can we not just access that? 

I recommend 
* Passing the object instead of just the ID to set service - the `name` attribute is available on the object and the ID can be set as it was
OR 
* Putting the services in a store and simply looking the item up by ID, this will be instant and not require additional requests